### PR TITLE
Implement "reset algorithm" for `<select>`.

### PR DESCRIPTION
**Pull request by frewsxcv**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13762)

### CODE REVIEW

**Comment by frewsxcv** [comment]
Suggestions for how to test this? I don't think there's anything in WPT that tests the "reset algorithm", but I might be wrong.

File: components/script/dom/htmlselectelement.rs [hunk 1]
```diff
@@ -84,6 +84,16 @@ impl HTMLSelectElement {
                            HTMLSelectElementBinding::Wrap)
     }
 
+    // https://html.spec.whatwg.org/multipage/#the-select-element:concept-form-reset-control
+    pub fn reset(&self) {
+        let node = self.upcast::<Node>();
+        for opt in node.traverse_preorder().filter_map(Root::downcast::<HTMLOptionElement>) {
```
Comment (Manishearth) [inline components/script/dom/htmlselectelement.rs]: Children, not descendants?

Comment (frewsxcv) [inline components/script/dom/htmlselectelement.rs]: Hah, I was already filing an issue before you even commented :P

Comment (frewsxcv) [inline components/script/dom/htmlselectelement.rs]: https://github.com/servo/servo/issues/13763

**Comment by Manishearth** [comment]
As for tests, add one that creates a simple form, resets it, and checks that the checkedness is correct again.

**Comment by frewsxcv** [comment]
Test case added in https://github.com/servo/servo/pull/13762/commits/957e1d8e7ad141d5ed95e0206f0f66326cd6f042.

**Comment by frewsxcv** [comment]
@Manishearth Everything look good here?

**Comment by Manishearth** [comment]
@bors-servo r+

oops, sorry this slipped through the cracks

**Pull request merged**